### PR TITLE
docs: initialize Sphinx Gallery

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Init for #2438
references: 
- https://sphinx-gallery.github.io/stable/getting_started.html
- https://stackoverflow.com/a/57216937

Remaining todo:
- _make thumbnail larger_ -> another PR
- [x] skip gallery for fast docs builds 
- ~add extra job/check for gallery~

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2508.org.readthedocs.build/en/2508/

### CODE REVIEW

**Comment by Borda** [comment]
asking about supporting `! pip install ...` in https://github.com/sphinx-gallery/sphinx-gallery/issues/129

**Comment by Borda** [comment]
sample: https://torchmetrics--2508.org.readthedocs.build/en/2508/gallery/image/spatial_correlation_coef.html#sphx-glr-gallery-image-spatial-correlation-coef-py

File: docs/source/conf.py [hunk 1]
```diff
@@ -146,13 +146,19 @@ def _set_root_image_path(page_path: str) -> None:
...
-nbsphinx_allow_errors = True
-nbsphinx_requirejs_path = ""
+sphinx_gallery_conf = {
+    "examples_dirs": os.path.join(_PATH_ROOT, "examples"),  # path to your example scripts
+    "gallery_dirs": "gallery",  # path to where to save gallery generated output
+    "filename_pattern": ".",
+    "ignore_pattern": r"__init__\.py",
+    "example_extensions": {".py"},
+    "line_numbers": True,
+    "promote_jupyter_magic": True,
```
Comment (Borda) [inline docs/source/conf.py:156]: This shall e able install...

**Review by SkafteNicki — APPROVED** [review]

**Comment by stancld** [comment]
Looks cool ❤️

File: requirements/_docs.txt [hunk 2]
```diff
@@ -23,4 +23,9 @@ pydantic > 1.0.0, < 3.0.0
 -r image.txt
 -r multimodal.txt
 -r text.txt
--r text_test.txt
+
+# Gallery extra requirements
+# --------------------------
+# todo: until this hase reolution - https://github.com/sphinx-gallery/sphinx-gallery/issues/1290
```
Comment (stancld) [inline requirements/_docs.txt:29]:
```suggestion
# todo: until this has resolution - https://github.com/sphinx-gallery/sphinx-gallery/issues/1290
```

**Review by stancld — APPROVED** [review]

**Pull request merged**